libs/react/src/lib/stepper/stepper.tsx

```diff
@@ -19,7 +19,8 @@ export function Stepper({
   const [stepper, data] = useStepper(props)
 
   useEffect(() => {
-    if (onChange && data.value) onChange(data.value)
+    if ( typeof data.value === 'undefined' ) return
+    if (onChange) onChange(data.value)
   }, [data.value])
```

Comment:
I think not in this case, onChange should not emit if only onChange is replaced and value stays the same? When onChange is replaced, the stepper component will anyways rerender and a new useEffect is created with a reference to the new onChange function

Reply:
Since it's common to do like this `<Stepper onChange={(value) => setState(value)} />` the onChange would emit every time that the outer component (perhaps a form with many other inputs) rerenders since the onChange is a new reference each time.

Reply:
Just talked to Atay who is using the stepper component, not specific to this pull request but specific to this discussion! The component calls onChange with the initial value on first render, and onChange is called twice for every value change when Strict Mode is enabled in development mode. So yet another reason to replace useEffect here 😄 When thinking about it, the logic can be moved into the extract stepper which is good since it will help the angular stepper too (once implemented)